test/firtool/verilog_equiv.fir

```diff
@@ -0,0 +1,299 @@
+; REQUIRES: yosys
+; RUN: split-file %s %t
+; RUN: firtool %t/test_mod.fir --format=fir -verilog -o %t/test_mod.fir.v
+; RUN: firtool %t/test_mod.fir --format=fir -mlir -o %t/test_mod.fir.mlir
+; RUN: firtool %t/test_mod.fir.mlir --lower-to-rtl -o %t/test_mod.rtl.mlir
+; RUN: firtool %t/test_unary.fir --format=fir -verilog -o %t/test_unary.fir.v
+; RUN: firtool %t/test_unary.fir --format=fir -mlir -o %t/test_unary.fir.mlir
+; RUN: firtool %t/test_prim.fir --format=fir -verilog -o %t/test_prim.fir.v
+; RUN: firtool %t/test_prim.fir --format=fir -mlir -o %t/test_prim.fir.mlir
```

Comment:
This looks like a test for the EmitVerilog changes above.  Please split these all out from the yosis integration patch and add a couple unit tests for the yosis driver under test/equiv-yosis/ or something.  Those should be written as .v files and shouldn't depend on the other circt tools.

The test for the EmitVerilog changes should be written in the style of unit tests like the other emitverilog tests.

Reply:
The mlir lines are residual of adding lower-to-rtl to verilog checks too.  Those fail right now, though I'll add the RUN lines for them.